src/bson.c

```diff
@@ -767,14 +796,66 @@ static const bson_visitor_t php_bson_visitors = {
 	{ NULL }
 };
 
+static bool map_element_matches_field_path(php_phongo_field_path_map_element* map_element, php_phongo_field_path* current)
+{
+	size_t i;
+
+	if (map_element->entry->size != current->size) {
+		return false;
+	}
+	for (i = 0; i < current->size; i++) {
+		if (strcmp(map_element->entry->elements[i], "$") == 0) {
+			continue;
+		}
+		if (strcmp(map_element->entry->elements[i], current->elements[i]) != 0) {
+			return false;
+		}
+	}
+	return true;
+}
+
+static php_phongo_field_path_map_element* map_find_field_path_entry(php_phongo_bson_state* state)
+{
+	size_t i;
+
+	/* Loop over all field path mappings, and for each, try to see whether it matches the current path */
+	for (i = 0; i < state->map.field_paths.size; i++) {
+		if (map_element_matches_field_path(state->map.field_paths.map[i], state->field_path)) {
```

Comment:
`map_element_matches_field_path()` may be a good candidate for an `inline` hint, where supported.

Reply:
I'll change that. But it would surprise me if compilers didn't already do that on their own.